packages/core/src/test/awsService/cloudWatchLogs/registry/liveTailSession.test.ts

```diff
@@ -65,6 +83,36 @@ describe('LiveTailSession', async function () {
         )
     })
 
+    it('startLiveTail sends command to CWL client with AbortSignal', async function () {
+        const sendCommandMock = sandbox.stub(CloudWatchLogsClient.prototype, 'send').callsFake(function () {
+            return buildLiveTailCommandOutput(testLogGroupArn)
+        })
+        const session = buildLiveTailSession({ type: 'all' }, undefined)
+        const stream = await session.startLiveTailSession()
+
+        assert.strictEqual(sendCommandMock.calledOnce, true)
+        assert.strictEqual(
+            sendCommandMock.calledWith(
+                sinon.match.any,
+                sinon.match({
+                    abortSignal: sinon.match.defined,
+                })
+            ),
+            true
+        )
+        assert.strictEqual(session.isAborted, false)
+        assert.deepEqual(stream, buildLiveTailCommandOutput(testLogGroupArn).responseStream)
+    })
+
+    it('stopLiveTail fires AbortSignal and destroys CWL client', function () {
+        const destroyClientSpy = sandbox.spy(CloudWatchLogsClient.prototype, 'destroy')
```

Comment:
why does this need a spy? test the actual state. the client can expose some (readonly) state if needed.

do not use mocks and spies except as a last resort. it looks like it's the first resort in this tests, instead of the last resort.

Reply:
I don't think there's a way to directly test the state of the underlying AWS SDK CWL client to check if its destroyed. Calling destroy on the Client doesn't stop it from being used from a future `send` call, it just destroys the underlying socket connections. There is no return value from `destroy` nor a `isDestroyed` (or something similar) to return its state.

Using a Spy here allows us to assert that a `.destroy()` call is made in a `stopLiveTailSession` call.